.github/workflows/package.yaml

```diff
@@ -68,9 +77,10 @@ jobs:
             [ "$VERSION" == "master" ] && VERSION=latest
             echo IMAGE_ID=$IMAGE_ID
             echo VERSION=$VERSION
-            docker tag $IMAGE_NAME $IMAGE_ID:$VERSION
-            docker push $IMAGE_ID:$VERSION
-          else
-            echo "Skipped on PRs"
-          fi
+            echo "::set-output name=IMAGE_TAG::$(
+              echo $IMAGE_ID:$VERSION
+            )"
 
+      - name: Build Docker Image and push to registry
+        if: ${{ github.ref }} =~ ^refs\/heads\/master|^refs\/tags\/v.*
+        run: docker buildx build --push --platform=linux/amd64,linux/arm64 --cache-from=type=local,src=/tmp/build-cache --tag ${{ steps.preperations.outputs.IMAGE_TAG }} --label "runnumber=${GITHUB_RUN_ID}" docker/
```

Comment:
wouldn't this build the image twice when it's on master or a version tag?

Reply:
ah it's all cached, disregard

Reply:
Correct it is cached from the previous steps. In theory you could run multiple builds in parallel on multiple machines (don't know how to do this in Github atm due to shared storage etc) and combine them in the final step and push it to the registry. Don't think this will offer much in term of time as buildx is already running these in parallel on a single machine.